tests/WatchTests.cs

```diff
@@ -355,11 +343,17 @@ public void TestWatchWithHandlers()
                 var events = new HashSet<WatchEventType>();
 
                 var watcher = listTask.Watch<V1Pod>(
-                    (type, item) => { events.Add(type); }
+                    (type, item) => {
+                        events.Add(type);
+                        eventsReceived.Signal();
```

Comment:
here might be some bad practice
https://www.jetbrains.com/help/resharper/2018.1/AccessToDisposedClosure.html

Reply:
In this case I think that rule may not be correct - the watcher is in the same parent scope as the closure and should be disposed before the wait handle?

Reply:
Agree.
but, in best practice, the code, `watcher.cs`, might be changed and those conditions would be false someday.

I suggest follow some strict static checking even it is not a bug.

Reply:
Yeah fair enough - that's a good reason 😊

Should make sure that the wait handle is disposed deterministically at some point though (perhaps at the end of the test?) - otherwise you have to rely on GC to avoid leaking native handles.

Reply:
Or I can move to Monitor instead of the CountdownEvent.

Reply:
How will you count the required number of events?